### PR TITLE
[breaking] pruning and reorganizing fields from protos, moving stuff to state

### PR DESCRIPTION
**Pull request by heavycrystal**
### ⚠️ This change can break existing CDC mirrors!

Performs several breaking changes to our protobufs:

1) Removes unused fields from `FlowConnectionConfigs`: `table_schema, src_table_id_name_mapping, table_name_schema_mapping, snapshot_sync_mode, cdc_sync_mode, push_batch_size, push_parallelism, metadata_peer`. `src_table_id_name_mapping` and `table_name_schema_mapping` are now moved to state, setup from `SetupFlow` and passed accordingly via more proto changes. Order shuffled around to group similar fields together.
2) Removed unnecessary generalization of `table_identifier`
3) Removed unused messages `EnsurePullabilityInput` and `EnsurePullabilityOutput`
4) Changed wording of `do_initial_copy` and `initial_copy_only` to better align with other options.
5) Removed `columns` from `TableSchema` and fixed related methods up.
6) `batch_size` had 3 different datatypes and was being modified in multiple places, making sure `batch_size` and `idle_timeout_seconds` has one source of truth.

### CODE REVIEW

File: flow/workflows/normalize_flow.go [hunk 1]
```diff
@@ -31,18 +31,20 @@ func NewNormalizeFlowExecution(ctx workflow.Context, state *NormalizeFlowState)
 
 func NormalizeFlowWorkflow(ctx workflow.Context,
 	config *protos.FlowConnectionConfigs,
+	options *protos.NormalizeFlowOptions,
```
Comment (serprex) [inline flow/workflows/normalize_flow.go:34]: normalize changes should be held off for #893

File: nexus/flow-rs/src/grpc.rs [hunk 2]
```diff
@@ -155,7 +155,7 @@ impl FlowGrpcClient {
             destination: Some(dst),
             flow_job_name: job.name.clone(),
             table_mappings,
-            do_initial_copy,
+            do_initial_snapshot: do_initial_copy,
```
Comment (serprex) [inline nexus/flow-rs/src/grpc.rs:158]: rename variable?

File: flow/connectors/utils/columns.go [hunk 3]
```diff
@@ -4,51 +4,27 @@ import (
 	"slices"
 
 	"github.com/PeerDB-io/peer-flow/generated/protos"
-	"golang.org/x/exp/maps"
 )
 
 func TableSchemaColumns(schema *protos.TableSchema) int {
```
Comment (serprex) [inline flow/connectors/utils/columns.go:9]: This file should probably be removed entirely

File: protos/flow.proto [hunk 4]
```diff
@@ -114,10 +101,12 @@ message SyncFlowOptions {
...
+  map<uint32, string> src_table_id_name_mapping = 4;
+  map<string, TableSchema> table_name_schema_mapping = 5;
 }
 
 message NormalizeFlowOptions {
-  int32 batch_size = 1;
+  map<string, TableSchema> table_name_schema_mapping = 1;
 }
 
 message LastSyncState {
```
Comment (serprex) [inline protos/flow.proto:112]: can get rid of `LastSyncState` we only use it to pass around `int64`

Comment (iskakaushik) [reply protos/flow.proto:112]: better to have it around in case we add more foelds and make it backwards compatible

Comment (serprex) [reply protos/flow.proto:112]: Can remove `last_synced_at`

File: flow/workflows/normalize_flow.go (hunk 1 above)
Comment (heavycrystal) [reply flow/workflows/normalize_flow.go:34]: needed to pass `TableNameSchemaMapping` in, was previously contained in config

File: flow/connectors/utils/columns.go (hunk 3 above)
Comment (heavycrystal) [reply flow/connectors/utils/columns.go:9]: will do in a later PR, will require a lot of changes

File: protos/flow.proto (hunk 4 above)
Comment (heavycrystal) [reply protos/flow.proto:112]: removed `last_synced_at`

File: nexus/flow-rs/src/grpc.rs (hunk 2 above)
Comment (heavycrystal) [reply nexus/flow-rs/src/grpc.rs:158]: done

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**